src/main/java/com/google/sps/data/SampleData.java

```diff
@@ -146,21 +147,50 @@ public static Student getStudentByEmail(String email) {
         return null;
     }
 
-    /** Deletes the given timeslot from the given tutor's availability array. */
+    /** Deletes the given timeslot from the given tutor's availability list. */
     public static void deleteAvailabilityByTimeRange(String email, TimeRange delete) {
         Tutor tutor = getTutorByEmail(email);
 
         for(TimeRange timeslot : tutor.getAvailability()) {
 
-            if(timeslot.getStart() == delete.getStart() && timeslot.getEnd() == delete.getEnd()) {
-                tutors.remove(tutor);
+            if (isTimeRangeEqual(delete, timeslot)) {
                 tutor.deleteAvailabilityByTimeRange(timeslot);
-                tutors.add(tutor);
                 break;
             }
         }
     }
 
+    /** Adds the given timeslot to the given tutor's availability list. */
+    public static void addAvailabilityByTutorEmail(String email, TimeRange timeslot) {
+        Tutor tutor = getTutorByEmail(email);
+        tutor.addAvailabilityByTimeRange(timeslot);
+    }
+
+    private static boolean isTimeRangeEqual(TimeRange timerange1, TimeRange timerange2) {
+        Calendar timerange1Calendar = timerange1.getDate();
```

Comment:
Instead of creating a new Calendar date, could you just compare start, end, and date of the two time ranges?

Reply:
I did this and it worked well! I had to create a new PR for this because of the cloud issues I was getting but I added this in there. I'll finalize that PR soon.